localstripe/resources.py

```diff
@@ -1602,6 +1602,37 @@ def _api_upcoming_invoice(cls, customer=None, subscription=None,
 
         return invoice
 
+    @classmethod
+    def _api_create_preview_invoice(cls, customer=None, subscription=None,
+                                    subscription_details=None):
+        try:
+            if subscription_details is not None:
+                assert type(subscription_details) is dict
+        except AssertionError:
+            raise UserError(400, 'Bad request')
+
+        details = subscription_details or {}
+        default_tax_rates = details.get('default_tax_rates')
+        items = details.get('items')
+        proration_date = details.get('proration_date')
+        trial_end = details.get('trial_end')
+
+        invoice = cls._get_next_invoice(
+            customer=customer,
+            subscription=subscription,
+            upcoming=True,
+            subscription_default_tax_rates=default_tax_rates,
+            subscription_items=items,
+            subscription_proration_date=proration_date,
+            subscription_trial_end=trial_end)
+
+        # Do not store this invoice
+        del store[cls.object + ':' + invoice.id]
```

Comment:
The Stripe docs says that an `upcoming` invoice can be retrieved via the `/v1/invoices/id` for 72 hours (cf https://docs.stripe.com/invoicing/preview), so maybe it would make more sense to keep these `upcoming` invoice in the store ?

Reply:
It would work for me. I'm curious about what @feliixx thinks!

Reply:
> I don't want to implement such "temporary" store in localstripe that would complicate things.

+1 not to implement a "temporary" store: we don't need to access the preview invoices via the `/v1/invoices/:id` route right now so it's not worth the extra complexity

> Also, please note that the /v1/invoices/:id route is not supported in localstripe right now so storing the invoice would not be useful.

Ok interesting, I wasn't aware of that ! 

> Maybe I can change the comment to says that real Stripe servers keep these invoices for a limited amount of time? What do you think?

Seems like the best solution :+1:

Reply:
Comment updated: [diff](https://github.com/adrienverge/localstripe/compare/019a82fa3d339d39a2b2af4215649445ecf02752..9e0c094f505ce04149ceca0d9c816b9d43c03497)